x/reward/keeper/msg_server_set_reward_test.go

```diff
@@ -44,7 +50,22 @@ func TestMsgSetRewards(t *testing.T) {
 	launchTriggeredChain.LaunchTriggered = true
 	launchTriggeredChainID := lk.AppendChain(sdkCtx, launchTriggeredChain)
 
-	err = bk.MintCoins(sdkCtx, types.ModuleName, moduleBalance.Add(newBalance...))
+	emptyBalanceLaunchID := lk.AppendChain(sdkCtx, sample.Chain(4, res.CoordinatorID))
+	k.SetRewardPool(sdkCtx, types.RewardPool{
+		LaunchID:            emptyBalanceLaunchID,
+		Coins:               emptyCoinsBalance,
+		LastRewardHeight:    100,
+		CurrentRewardHeight: 30,
+	})
+	zeroRewardHeightLaunchID := lk.AppendChain(sdkCtx, sample.Chain(5, res.CoordinatorID))
+	k.SetRewardPool(sdkCtx, types.RewardPool{
+		LaunchID:            zeroRewardHeightLaunchID,
+		Coins:               zeroRewarHeightBalance,
+		LastRewardHeight:    100,
+		CurrentRewardHeight: 30,
+	})
+
+	err = bk.MintCoins(sdkCtx, types.ModuleName, moduleBalance)
```

Comment:
I understand from the comment above that we must initialize coins object to initialize reward pools and mint here.

My suggestion would be to add in the test case structure a `InitialRewardPool`. When defined, the logic of initializing it and minting tokens would be performed in the body of the TDT.
It would become in my mind more in the idea of TDT where the each test case define (initialState, input, expectedOutput)

Reply:
Maybe we could also think in the future of even refactoring the coordinator part above where you provide the coordinator and associated balance in test cases

Reply:
I agree!! I think this applies to all message test methods. We can define the initial keeper state and the expected output state. This can be a new issue, for we investigate the better way to do it.

Reply:
https://github.com/tendermint/spn/issues/556